src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,133 @@
+import elasticsearch from 'elasticsearch';
+import { get, set, isEmpty, cloneDeep, pick } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Boom from 'boom';
+
+import filterHeaders from './filter_headers';
+import { parseConfig } from './parse_config';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    this._client = this.createClient();
+    this._noAuthClient = this.createClient({ auth: false });
+
+    return this;
+  }
+
+  /**
+   * callWithRequest
+   *
+   * Performs a call to ES, passing through whitelisted headers in the request
+   *
+   * @param {Object|undefined} req - The request object
+   * @param {string} endpoint
+   * @param {Object} clientParams
+   * @param {Object} options
+   * @param {boolean} options.wrap401Errors
```

Comment:
documenting the fields in `options` object or documenting the input parameters to the function?  I get the input parameters (and I actually really like them, as per above), it's just the very last one that doesn't look right to me because it's not an input parameter, and I don't think thats the right way to document `options`.  Instead I would expect something like this:

/**
 * @typedef {Object} RequestOptions
 * @property {boolean} wrap401Errors
 */

/** 
 * ... other stuff
 * @param {Object} clientParams
 * @param {RequestOptions} options
 */
callWithRequest = ....

Anyway, fine whatever you decide, I'm just curious if this is an official JSDoc way of doing the above that I was unaware of.